### ISSUE TITLE
If settings.BASE_DIR is a string, django-components errors when loading the templates

### ISSUE THREAD

**Issue by alexgmin** [EmilStenstrom/django-components#381]
In my settings file I had this as `BASE_DIR`
`BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))`
Which was later used for this in `TEMPLATES`:
`'DIRS': [os.path.join(BASE_DIR, 'templates')],`

When I added django-components to the project this error happened
```
  File "/foo/lib/python3.11/site-packages/django_components/template_loader.py", line 18, in get_dirs
    path = (settings.BASE_DIR / component_dir).resolve()
            ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for /: 'str' and 'str'
```

The solution is to change the `BASE_DIR`to `BASE_DIR = Path(__file__).resolve().parent.parent.parent`, which is what https://github.com/EmilStenstrom/django-components/issues/290 does in the example settings.

Since `BASE_DIR` is not an official Django setting, just something mentioned [in some places of the documentation](https://docs.djangoproject.com/en/5.0/search/?q=base_dir), the library documentation should explain what the setting is and what value is expected to have (a `pathlib.Path` instance)

**Comment by EmilStenstrom** [comment]
Hmm... maybe we should handle the case that it's a string too. I think wrapping BASE_DIR in Path is safe no matter if it's a Path or not from the start? Mind adding that as a PR?

**Comment by alexgmin** [comment]
I can confirm that wrapping a Path again is safe.
```python
>>> foo = pathlib.Path('foo')
>>> bar = pathlib.Path(foo)
>>> foo == bar
True
```

I'll try to create a PR this week for the code. I'll also write a test. If someone else wants to do it instead I don't mind.
However I think the readme should have a section about the BASE_DIR setting but I don't really know how to explain it.

**Comment by EmilStenstrom** [comment]
@spapas this is a hobby project for everyone here! If you have the bandwidth to wrap our Base_dir usage with Path() we will happily merge it.

**Comment by spapas** [comment]
@EmilStenstrom unfortunately not right now, I'm to pressured by other stuff... I've resolved the issue by staying on 0.34.1 for now.
